java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestNavigationUtils.java

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+* Copyright (c) 2021 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package com.microsoft.java.test.plugin.util;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.search.IJavaSearchConstants;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.SearchPattern;
+import org.eclipse.jdt.core.search.TypeNameRequestor;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Utils for test navigation features
+ */
+public class TestNavigationUtils {
+    /**
+     * find test or test target according to the given java source file uri.
+     * @param arguments arguments
+     * @param monitor monitor
+     * @return the search result for test navigation
+     * @throws JavaModelException
+     */
+    public static List<TestFindResult> findTestOrTarget(List<Object> arguments, IProgressMonitor monitor)
+            throws JavaModelException {
+        if (arguments == null || arguments.size() < 2) {
+            throw new IllegalArgumentException("Wrong arguments passed to findTestOrTarget().");
+        }
+        final String typeUri = (String) arguments.get(0);
+        final ICompilationUnit unit = JDTUtils.resolveCompilationUnit(typeUri);
+        if (unit == null) {
+            JUnitPlugin.logError("Failed to resolve compilation unit from " + typeUri);
+            return null;
+        }
+        final boolean goToTest = (boolean) arguments.get(1);
```

Comment:
what's "false" reserved for?

Reply:
It's reserved for `test code -> business code`